src/qt/modaloverlay.cpp

```diff
@@ -170,3 +189,21 @@ void ModalOverlay::closeClicked()
     showHide(true);
     userClosed = true;
 }
+
+void ModalOverlay::pauseClicked()
+{
+    Q_EMIT requestVerificationPauseOrResume();
+}
+
+void ModalOverlay::setPauseResumeState(bool pauseActive)
+{
+    verificationPauseActive = pauseActive;
+    updatePauseState(pauseActive);
+}
+
+void ModalOverlay::updatePauseState(bool pauseActive)
+{
+    ui->labelNumberBlocksRequested->setText((pauseActive ? "Finish downloading blocks": "Blocks requested from peers"));
```

Comment:
Don't see the need to change label, maybe a generic `Downloading: %d blocks`?

Reply:
I think the currents PR label switch makes it more clear what happens in the background (pause syncing will not lead to immediate network download halt).